### PR TITLE
[Threshold] histogram and slider problems

### PR DESCRIPTION
**Pull request by mathildemerle**
Solves https://github.com/medInria/medInria-public/issues/587 i.e in the Threshold an image toolbox:
 * click on the Histogram button: no more crash
 * the sliders were missing ranges: they can be used
 * click on the Histogram button: it takes so long to open (but that's normal) that i put the toolbox in wait mode

BTW, i saw an other problem with this toolbox. If you Run twice, there are some buttons and display problems, as if something is running in loop somewhere. I'll investigate too. And also i'll do the PR on master at the end.

:m:

### CODE REVIEW

File: src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp [hunk 1]
```diff
@@ -276,30 +276,35 @@ itkFiltersToolBox::itkFiltersToolBox(QWidget *parent)
     d->thresholdFilterWidget = new QWidget(this);
 
     d->thresholdFilterValue = new medDoubleParameterL( tr("Value"), this);
-    d->thresholdFilterValue->setRange ( std::numeric_limits<double>::lowest() , std::numeric_limits<double>::max() );
-    d->thresholdFilterValue->setValue ( itkFiltersThresholdingProcess::defaultThreshold );
-    d->thresholdFilterValue->setObjectName("thresholdValue");
+    d->thresholdFilterValue->setToolTip(tr("Only keep values according to the threshold type"));
+    d->thresholdFilterValue->setValue(itkFiltersThresholdingProcess::defaultThreshold);
+    d->thresholdFilterValue->setRange(-10000, 10000);
```
Comment (Florent2305) [inline src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp:281]: Pourquoi ne pas garder: `std::numeric_limits<double>::lowest() , std::numeric_limits<double>::max()` qui est algorithmiquement très bien

File: src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp [hunk 2]
```diff
@@ -276,30 +276,35 @@ itkFiltersToolBox::itkFiltersToolBox(QWidget *parent)
...
+    d->thresholdFilterValue->setObjectName("thresholdFilterValue");
 
     d->thresholdLowerValue = new medDoubleParameterL( tr("Lower Value"), this);
-    d->thresholdLowerValue->setRange ( std::numeric_limits<double>::lowest() , std::numeric_limits<double>::max() );
+    d->thresholdLowerValue->setToolTip(tr("Choose the boundaries of your threshold"));
+    d->thresholdLowerValue->setRange(-10000, 10000);
```
Comment (Florent2305) [inline src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp:286]: idem

File: src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp [hunk 3]
```diff
@@ -276,30 +276,35 @@ itkFiltersToolBox::itkFiltersToolBox(QWidget *parent)
...
     d->thresholdLowerValue->setValue ( itkFiltersThresholdingProcess::defaultLower );
     d->thresholdLowerValue->setObjectName("lowerValue");
     d->thresholdLowerValue->setDecimals(10);
+    d->thresholdLowerValue->setObjectName("thresholdLowerValue");
 
     d->thresholdUpperValue = new medDoubleParameterL( tr("Upper Value"), this);
-    d->thresholdUpperValue->setRange ( std::numeric_limits<double>::lowest() , std::numeric_limits<double>::max() );
+    d->thresholdUpperValue->setToolTip(tr("Choose the boundaries of your threshold"));
+    d->thresholdUpperValue->setRange(-10000, 10000);
```
Comment (Florent2305) [inline src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp:294]: idem

**Comment by mathildemerle** [comment]
This PR can be reviewed, the found bug has been declared on https://github.com/medInria/medInria-public/issues/593

File: src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp [hunk 4]
```diff
@@ -803,9 +814,7 @@ void itkFiltersToolBox::updateClutEditorView()
     {
         QList<medClutEditorVertex*>& vertices = d->clutEditor->getScene()->table()->vertices();
         if ( vertices.size()==0 )
-        {
```
Comment (Florent2305) [inline src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp:806]: { }?

File: src/plugins/legacy/medVtkView/medVtkView.cpp [hunk 5]
```diff
@@ -223,12 +231,16 @@ QWidget* medVtkView::viewWidget()
     return d->viewWidget;
 }
 
+QMainWindow* medVtkView::mainWindow()
```
Comment (Florent2305) [inline src/plugins/legacy/medVtkView/medVtkView.cpp:234]: Pourquoi?

File: src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp (hunk 1 above)
Comment (mathildemerle) [reply src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp:281]: I forgot to explain that these numeric limits values were not allowed in these parameters. They were blocking the range, and we couldn't use the sliders

File: src/plugins/legacy/medVtkView/medVtkView.cpp (hunk 5 above)
Comment (mathildemerle) [reply src/plugins/legacy/medVtkView/medVtkView.cpp:234]: This is needed to get the main Window, from the medVtkView. The histogram is built on it

File: src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp (hunk 1 above)
Comment (Florent2305) [reply src/plugins/legacy/itkFilters/itkFiltersToolBox.cpp:281]: ok

**Pull request merged**